internal/search/query/types.go

```diff
@@ -344,6 +344,16 @@ func (q Q) Repositories() (repos []string, negatedRepos []string) {
 	return repos, negatedRepos
 }
 
+func (q Q) Dependencies() (dependencies []string) {
+	VisitField(q, FieldDependencies, func(value string, negated bool, _ Annotation) {
+		if !negated {
+			dependencies = append(dependencies, value)
+			return
+		}
+	})
+	return dependencies
+}
+
```

Comment:
```suggestion
	VisitPredicate(q, func(field, name, value string) {
		if field == FieldRepo && (name == "dependencies" || name == "deps") {
			dependencies = append(dependencies, value)
			return
		}
	})
	return dependencies
}
```

Easy, now this is all we have to do. It's possible we can do this more robustly by using the `Predicate` type you minted so we're not matching on the `"dependencies"` and `"deps"` strings directly, but this suffices for now.